js/src/views/Contract/Events/Event/event.js

```diff
@@ -0,0 +1,183 @@
+// Copyright 2015, 2016 Ethcore (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+import BigNumber from 'bignumber.js';
+import moment from 'moment';
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+import { fetchBlock, fetchTransaction } from '../../../../redux/providers/blockchainActions';
+import { IdentityIcon, IdentityName, Input, InputAddress } from '../../../../ui';
+
+import styles from '../../contract.css';
+
+class Event extends Component {
+  static contextTypes = {
+    api: PropTypes.object.isRequired
+  }
+
+  static propTypes = {
+    event: PropTypes.object.isRequired,
+    blocks: PropTypes.object,
+    transactions: PropTypes.object,
+    isTest: PropTypes.bool,
+    fetchBlock: PropTypes.func.isRequired,
+    fetchTransaction: PropTypes.func.isRequired
+  }
+
+  componentDidMount () {
+    this.retrieveTransaction();
+  }
+
+  render () {
+    const { event, blocks, transactions, isTest } = this.props;
+
+    const block = blocks[event.blockNumber.toString()];
+    const transaction = transactions[event.transactionHash] || {};
+    const classes = `${styles.event} ${styles[event.state]}`;
+    const url = `https://${isTest ? 'testnet.' : ''}etherscan.io/tx/${event.transactionHash}`;
```

Comment:
Couldn't we have this URL constructor more DRY ? If etherscan ever goes down, or change its URL structure, it would be good to only modify it in one place whether than in every component